lib/src/themes/feature_resolver.dart

```diff
@@ -0,0 +1,59 @@
+import '../../vector_tile_renderer.dart';
+import 'theme_layers.dart';
+
+/// Resolver for resolving the [VectorTileFeature]s a [ThemeLayer] should
+/// render.
+abstract class ThemeLayerFeatureResolver {
+  /// Resolves and returns the [VectorTileFeature]s that the given [themeLayer]
+  /// should render.
+  Iterable<ResolvedThemeLayerFeature> resolveFeatures(DefaultLayer themeLayer);
+}
+
+/// Default implementation of [ThemeLayerFeatureResolver] that resolves the
+/// features contained in a specific [tileset] for a [ThemeLayer].
+class DefaultThemeLayerFeatureResolver implements ThemeLayerFeatureResolver {
+  DefaultThemeLayerFeatureResolver(this.tileset);
+
+  /// The [Tileset] from which to resolves [VectorTileFeature]s.
+  final Tileset tileset;
+
+  @override
+  Iterable<ResolvedThemeLayerFeature> resolveFeatures(
+    DefaultLayer themeLayer,
```

Comment:
rather than passing a `DefaultLayer`, how about passing a `TileLayerSelector`? That would avoid limiting the resolver to `DefaultLayer`

Reply:
Sounds good.